statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,698 @@
+        See Parameters.
+    estimation_kwds : dict-like or None
+        See Parameters.
+    join_method : function or None
+        See Parameters.
+
+    Examples
+    --------
+
+    Notes
+    -----
+    """
+
+    def __init__(self, data_generator, partitions, model_class=None,
+                 init_kwds=None, fit_kwds=None, estimation_method=None,
+                 estimation_kwds=None, join_method=None,
+                 join_kwds=None):
+
+        self.data_generator = data_generator
+        self.partitions = partitions
+
+        if model_class is None:
+            self.model_class = OLS
+        else:
+            self.model_class = model_class
+
+        if init_kwds is None:
+            self.init_kwds = {}
+        else:
+            self.init_kwds = init_kwds
+
+        if fit_kwds is None:
+            self.fit_kwds = {}
+        else:
+            self.fit_kwds = fit_kwds
+
+        if estimation_method is None:
+            self.estimation_method = _est_regularized_debiased
+        else:
+            self.estimation_method = estimation_method
+
+        if estimation_kwds is None:
+            self.estimation_kwds = {}
+        else:
+            self.estimation_kwds = esitmation_kwds
+
+        if join_method is None:
+            self.join_method = _join_debiased
+        else:
+            self.join_method = join_method
+
+        if join_kwds is None:
+            self.join_kwds = {}
+        else:
+            self.join_kwds = join_kwds
+
+
+#    def fit_partition(self, pnum, endog, exog):
+#        """handles the model fitting for each machine.
+#
+#        Parameters
+#        ----------
+#        pnum : scalar
+#            index of current partition.
+#        endog : array-like
+#            endogenous data for current partition.
+#        exog : array-like
+#            exogenous data for current partition.
+#
+#        Returns
+#        -------
+#        estimation_method result.  For the default,
+#        _est_regularized_debiased, a tuple.
+#        """
+#
+#        model = self.model_class(endog, exog, **self.init_kwds)
+#        results = self.estimation_method(model, pnum, self.partitions,
+#                                         self.fit_kwds,
+#                                         **self.estimation_kwds)
+#        return results
+
+
+    def fit_distributed(self, parallel_method="sequential"):
+        """Performs the distributed estimation using the corresponding
+        DistributedModel
+
+        Parameters
+        ----------
+        parallel_method : str
+            type of distributed estimation to be used, currently
+            "sequential", "joblib" and "dask" are supported.
+
+        Returns
+        -------
+        join_method result.  For the default, _join_debiased, it returns a
+        p length array.
+        """
+
+        if parallel_method == "sequential":
+            return self.fit_dist_sequential()
+
+        elif parallel_method == "joblib":
+            try:
```

Comment:
No need for this `try` block because the statsmodels parallel tool will do the check again.